pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFileUploadNotifications.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.ddp.studybuilder.task;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.db.dao.MailTemplateDao;
+import org.broadinstitute.ddp.db.dao.MailTemplateRepeatableElementDao;
+import org.broadinstitute.ddp.db.dto.MailTemplateDto;
+import org.broadinstitute.ddp.db.dto.MailTemplateRepeatableElementDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Task to update existing singular study by specifying newly added mail template
+ */
+@Slf4j
+public class SingularFileUploadNotifications implements CustomTask {
+    private static final String RECORD_FILE = "emails/file-upload-notification/file-record-en.html";
+    private static final String BODY_FILE   = "emails/file-upload-notification/en.html";
+    private static final String STUDY_GUID  = "singular";
+
+    protected Config dataCfg;
+    protected Path cfgPath;
+    protected Config cfg;
+    protected Config varsCfg;
+
+    @Override
+    public void init(final Path cfgPath, final Config studyCfg, final Config varsCfg) {
+        if (!studyCfg.getString("study.guid").equals(STUDY_GUID)) {
+            throw new DDPException("This task is only for the " + STUDY_GUID + " study!");
+        }
+
+        final var file = cfgPath.getParent().toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+    }
+
+    @Override
+    public void run(final Handle handle) {
+        log.info("TASK:: {}", SingularFileUploadNotifications.class.getSimpleName());
+
+        final var mailTemplateId = insertMailTemplate(handle);
+        insertMailTemplateRepeatableElement(handle, mailTemplateId);
+        log.info("Mail template inserted");
+
+        handle.attach(SqlHelper.class).updateStudyMailTemplate(cfg.getString("study.guid"), mailTemplateId);
+        log.info("Study {} mail template was updated", cfg.getString("study.guid"));
```

Comment:
I would suggest returning early if `notification_mail_template_id` is not `null`, or add some clearer error handling. As it stands this patch will fail with a fatal SQL error due to the unique constraints on the `mail_template_repeatable_element` if run multiple times.

Reply:
Well, reviewed once again. I decided to remove the template's name. We don't use it at all. So, now each time we run the patch it will create a new mail template instance and update the study's reference to the newly created one. I believe it corresponds to the logic we use for study reimport.

Reply:
This is not a study reimport, though- this is being run against a study that has previously been configured. This is now leaving orphaned values in the database.

Reply:
So, what is your suggestion?
1. Unlink study & template
2. Delete template
3. Insert template

Reply:
Or! We can insert new template and then delete everything unlinked.

Reply:
My perspective on this patch is this is being run prior to any study configuration being implemented for handling templates in the HOCON files, so when this patch is run, I would expect `notification_mail_template_id` to be `null`.

If `notification_mail_template_id` is not `null`, it's likely the patch has already been run, or this patch has been superseded by study configuration, in which case I would suggest just logging that a template has already been added, and successfully returning early so we don't overwrite the potentially newer changes.

Reply:
Anyway, I implemented the deletion. But to be honest, I don't think it's an important thing in our case. How many patches are we going to apply? 10? 100? It will not affect the performance but will keep the changes history.

Reply:
So, the current implementation should not leave any garbage.